bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -869,6 +897,17 @@ public boolean setText(String html, boolean trusted) {
 }
 
 private boolean setWebpageData(String url, String postData, String[] headers, String html) {
+	// If the create() method hasn't finished executing, queue the call to wait for the browser to finish initializing
+	if (!initializationProcessFinished.isDone()) {
+		final String fUrl = url;
+		browser.getDisplay().asyncExec(() -> {
+			if (waitForInitialization()) {
+				setWebpageData(fUrl, postData, headers, html);
+				browserResize(new Event());
+			}
+		});
+		return true;
+	}
```

Comment:
Unfortunately, and in my opinion, this leads to an API break of `Browser#setUrl()` and `Browser#setText()`. Those methods' specification say that they return whether setting the URL or text was successful, but with this change the result depends on whether the actual operation is performed asynchronously or not. I.e., the browser will always return "true" if has not been completely initialized when calling this method while it may return "false" in other cases. Or is there some implicit guarantee that the asynchronous execution cannot happen when calling the method from outside consumers?

Reply:
Well, you are right. But the problem is that, webview serializes execution and if there's a call to setUrl or string while the initialization is still in process, then it would wait for the initialization and then execute. And we do this asynchrnously just to make sure that there's no deadlock. since, this scenario will only happen when a new browser is created in a webview callback and wants to set a new page on the new browser. In that case they need to execute asyncly to let the callback finish and then execute setup of the browser followed by setWebPage.

And yes atleast from the architecture, there's a guarantee that async execution won't happen if the browser is already setup. since setup is only delayed when a new browser is created in a callback, since the deadlock condition would be true.